libs/model/src/community/GetCommunities.query.ts

```diff
@@ -0,0 +1,248 @@
+import { InvalidState, type Query } from '@hicommonwealth/core';
+import * as schemas from '@hicommonwealth/schemas';
+import { QueryTypes } from 'sequelize';
+import { z } from 'zod';
+import { models } from '../database';
+
+export function GetCommunities(): Query<typeof schemas.GetCommunities> {
+  return {
+    ...schemas.GetCommunities,
+    auth: [],
+    secure: false,
+    body: async ({ payload }) => {
+      // TODO: constrain order_by
+      const {
+        base,
+        tag_ids,
+        include_node_info,
+        stake_enabled,
+        has_groups,
+        cursor,
+        limit,
+        order_by,
+        order_direction,
+      } = payload;
+      if (
+        order_by &&
+        order_by !== 'profile_count' &&
+        order_by !== 'thread_count'
+      ) {
+        throw new InvalidState(
+          'Invalid ordering: must be profile_count or thread_count.',
+        );
+      }
```

Comment:
The `order_by` validation should be part of the zod input schema. Can use `z.union`.

The comment at L13 can be removed

Reply:
Addressed.